File: rich_game_mechnism/src/com/tw/core/commands/SellEstate.java
package com.tw.core.commands;

import com.tw.core.Player;
import com.tw.core.places.Estate;
import com.tw.core.responses.Response;

/**
 * Created by pzzheng on 11/17/16.
 */
public class SellEstate implements Command {
    private Estate estate;

    public SellEstate(Estate estate) {
        this.estate = estate;
    }

    @Override
    public Player.Status execute(Player player) {
        if(player.getAsests().hasEstate(estate)){
            player.getAsests().addFund(estate.getEmptyPrice() * (estate.getLevel().ordinal() + 1) * 2);
            player.getAsests().removeEstate(estate);
            estate.beEmpty();
        }
        return player.waitForCommand();

    }

    @Override
    public Player.Status respond(Response response, Player player) {
        return null;
    }
}

File: ruby/rich_game_ruby/src/commands/sell_estate.rb
require_relative 'command'
class SellEstate < Command
  def initialize(estate)
    @estate = estate
  end

  def execute(player)
    if(player.asset.hasEstate(@estate)) then
      player.asset.sellEstate(@estate)
      @estate.sell
    end
    return player.waitForCommand
  end


end